xmrstak/backend/cpu/minethd.cpp

```diff
@@ -356,25 +356,35 @@ void minethd::work_main()
 			    either because of network latency, or a socket problem. Since we are
 			    raison d'etre of this software it us sensible to just wait until we have something*/
 
-			while (globalStates::inst().inst().iGlobalJobNo.load(std::memory_order_relaxed) == iJobNo)
+			while (globalStates::inst().iGlobalJobNo.load(std::memory_order_relaxed) == iJobNo)
 				std::this_thread::sleep_for(std::chrono::milliseconds(100));
 
 			consume_work();
 			continue;
 		}
 
+		size_t nonce_ctr = 250;
```

Comment:
Could it be that this must be `size_t nonce_ctr = 4096;` because 4096 nonce's are allocated.

Reply:
ahhh I get it, the nonce is only update if the stats get updated and this will be every 16 hashes. In this case nonce_ctr must be  `size_t nonce_ctr = 256;`